### PR TITLE
chore: avoid building tests and examples in uring project

### PR DESCRIPTION
**Pull request by romange**
Also, check socket on shutdown.

### CODE REVIEW

File: util/fibers/listener_interface.cc [hunk 1]
```diff
@@ -348,7 +348,7 @@ uint32_t ListenerInterface::GetMaxClients() const {
 }
 
 void Connection::Shutdown() {
-  auto ec = socket_->Shutdown(SHUT_RDWR);
+  auto ec = CHECK_NOTNULL(socket_)->Shutdown(SHUT_RDWR);
```
Comment (chakaz) [inline util/fibers/listener_interface.cc:351]: Just making sure - this is unrelated to the third_pary.cmake changes you've made above right?

**Review by chakaz — APPROVED** [review]

File: util/fibers/listener_interface.cc (hunk 1 above)
Comment (romange) [reply util/fibers/listener_interface.cc:351]: sorry, upodated the PR description. merging
